services/brig/src/Brig/API/User.hs

```diff
@@ -176,14 +176,19 @@ createUser new@NewUser{..} = do
     -- Handle e-mail activation
     edata <- if emailInvited || teamEmailInvited
             then return Nothing
-            else fmap join . for emKey $ \ek -> do
-                timeout <- setActivationTimeout <$> view settings
-                edata   <- lift $ Data.newActivation ek timeout (Just uid)
-                Log.info $ field "user"            (toByteString uid)
-                         . field "activation.key"  (toByteString $ activationKey edata)
-                         . field "activation.code" (toByteString $ activationCode edata)
-                         . msg (val "Created email activation key/code pair")
-                return $ Just edata
+            else fmap join . for emKey $ \ek -> case newUserEmailCode of
+                Nothing -> do
+                    timeout <- setActivationTimeout <$> view settings
+                    edata   <- lift $ Data.newActivation ek timeout (Just uid)
+                    Log.info $ field "user"            (toByteString uid)
+                             . field "activation.key"  (toByteString $ activationKey edata)
+                             . field "activation.code" (toByteString $ activationCode edata)
+                             . msg (val "Created email activation key/code pair")
```

Comment:
Do we actually make use of this log data, or could we not log the actual activation key/codes ?

Reply:
No need to log the `code`, but will keep the log for the activation key (hash representation of the actual key).
But good catch, worthwhile removing